src/session_impl.cpp

```diff
@@ -5641,8 +5629,14 @@ namespace aux {
 			m_dht->add_node(n);
 		}
 		m_dht_nodes.clear();
+		m_dht_nodes.shrink_to_fit();
 
-		m_dht->start(startup_state, std::bind(&on_bootstrap, std::ref(m_alerts)));
+		auto cb = [this]
+		{
+			if (m_alerts.should_post<dht_bootstrap_alert>())
+				m_alerts.emplace_alert<dht_bootstrap_alert>();
+		};
+		m_dht->start(std::bind(cb));
```

Comment:
I think the ``bind`` here should be removed. I assume that if you found it to not work without the bind, there's some argument passed in to the callback that the lambda need to declare. ``std::bind`` will just take any argument ad throw it away.